### ISSUE TITLE
ContextMenuArea not working with SwingPanel

### ISSUE THREAD

**Issue by sal0max** [JetBrains/compose-multiplatform#4915]
**Describe the bug**
I tried to add a ContextMenuArea around a SwingPanel and to catch right clicks.

**Affected platforms**
- Desktop (Windows, Linux, macOS)

**Versions**
- Libraries:
  - Compose Multiplatform version: 1.6.10
- Kotlin version: 2.0.0
- OS version(s) (required for Desktop and iOS issues): mac OS 14.5 (23F79)
- OS architecture (x86 or arm64): arm64
- JDK (for desktop issues): openjdk 17.0.11 

**To Reproduce**

1. Run this code snippet:
   ```kt
   ContextMenuArea(
       items = {
       listOf(
           ContextMenuItem("User-defined Action") { },
           ContextMenuItem("Another user-defined action") { }
       )
   }) {
       SwingPanel(
           modifier = Modifier.fillMaxSize(),
           factory = { JTextField() }
       )
   }
   ```
3. Right-click inside the JTextField
4. No popup will open

**Expected behavior**
A popup should appear

**Comment by MatkovIvan** [comment]
Things in this case are complicated. The two main problem areas are:
1. Draw something above `SwingPanel`
2. Intercept input above `SwingPanel`

(1) is solved by https://github.com/JetBrains/compose-multiplatform-core/pull/915 but available only under a flag due to [some issues](https://github.com/JetBrains/compose-multiplatform/issues/4941).
(2) with (1) in should work as-is on WIndows, but for macOS requires [a workaround](https://github.com/JetBrains/compose-multiplatform-core/pull/915#issuecomment-1954155894) to redispatch Swing/AWT events to Compose is requried

Summarizing all points above it will be something like this - https://gist.github.com/MatkovIvan/11e08183b80f3a31b218a093a4ce2a9d

So it's possible to do things in this way, but I'd recommend avoid such situation if possible. Closing since it's already tracked in a couple of issues and I provided a way how to solve the problem
